big-o-notation/big-o.js

```diff
@@ -0,0 +1,44 @@
+/* exported uniqueQuadratic, uniqueLinear */
+
+function uniqueLinear(words) {
+  const seen = {};                  // 1 * 1 = O(1)
+  const unique = [];                // 1 * 1= O(1)
+  for (
+    let i = 0;                      // 1 * 1 = O(1)
+    i < words.length;               // 2 * n = O(n)
+    i++                             // 2 * n = O(n)
+  ) {
+    const word = words[i];          // 2 * n = O(n)
+    if (!seen[word]) {              // 2 * n = O(n)
+      seen[word] = true;
+      unique[unique.length] = word; // 3 * n = O(n)
+    }
+  }
+  return unique;
+} // Big O Notation for uniqueLinear: O(n)
+
+function uniqueQuadratic(words) {
+  const unique = [];                // 1 * 1 = O(1)
+  for (
+    let i = 0;                      // 1 * 1 = O(1)
+    i < words.length;               // 2 * n = O(n)
+    i++                             // 2 * n = O(n)
+  ) {
+    const word = words[i];          // 2 * n = O(n)
+    let isUnique = true;            // 1 * n = O(n)
+    for (
+      let c = 0;                    // 1 * n^2 = O(n^2)
```

Comment:
The initialization expression for the loop will only run one time per word, so `1 * n = O(n)`